src/backend/commands/analyze.c

```diff
@@ -713,8 +713,10 @@ do_analyze_rel(Relation onerel, VacuumParams *params,
 	 * statistics target. So we may need to sample more rows and then build
 	 * the statistics with enough detail.
 	 */
-	minrows = ComputeExtStatisticsRows(onerel, attr_cnt, vacattrstats);
```

Comment:
only check `IS_SINGLENODE` is fine?

Reply:
QD and  SINGLENODE need to calc minrows.